#!/usr/bin/env python
PACKAGE = "pwm_dev"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("kp_v",    double_t,    0, "Proportional Coefficient", 1.0, 0.0, 10.0)
gen.add("ki_v",    double_t,    0, "Integral Coefficient", 0.0, 0.0, 10.0)
gen.add("kd_v",    double_t,    0, "Derivative Coefficient", 0.0, 0.0, 10.0)

gen.add("kp_w",    double_t,    0, "Proportional Coefficient", 1.0, 0.0, 10.0)
gen.add("ki_w",    double_t,    0, "Integral Coefficient", 0.0, 0.0, 10.0)
gen.add("kd_w",    double_t,    0, "Derivative Coefficient", 0.0, 0.0, 10.0)

gen.add("max_i", double_t,    0, "Max Integral Accumulation", 5.0, 0.0, 10.0)
gen.add("max_v", double_t,    0, "Max Linear Velocity (m/s)", 1.3, 0.0, 2.7)
gen.add("max_w", double_t,    0, "Max Angular Velocity (rad/s)", 1.4, 0.0, 5.0)

exit(gen.generate(PACKAGE, "dynamic_tutorials", "PID"))
